File: learn-pr/azure/handle-transient-errors-in-your-app/src/node-chatapp-retry/retryPolicy.js
var method = retryPolicy.prototype;
var config = require('./appsettings.json');
var sleep = require('system-sleep');
const shared = new Int32Array(new ArrayBuffer(4));

// Constructor
function retryPolicy() {
    this._currentTries = 0;
}

method.checkRetries = function() {
    this._currentTries = this._currentTries + 1;
    console.log('Retrying: ' + this._currentTries);

    // Use a delay if this isn't the first try
    if (this._currentTries != 1)
    {
        sleep(config.delay);
    }

    if (this._currentTries < config.number_of_retries) {
        return true;
    } else {
        return false;
    }
};

// export the class
module.exports = retryPolicy;

File: learn-pr/azure/handle-transient-errors-in-your-app/src/csharp-chatapp-retry.1/RetryPolicy.cs
using System;
using Microsoft.Extensions.Configuration;
using System.Threading;
using System.Diagnostics;
using System.IO;


namespace csharp_chatapp_retry
{
    public class RetryPolicy
    {
        private int currentTries = 0;
        static public IConfiguration configuration { get; set; }

        /// <summary>
        /// Constructor - reads config for number of retries and delay
        /// </summary>
        public RetryPolicy()
        {
            ConfigurationBuilder configurationBuilder = new ConfigurationBuilder();
            configurationBuilder.SetBasePath(Directory.GetCurrentDirectory());
            configurationBuilder.AddJsonFile("appsettings.json");
            configuration = configurationBuilder.Build();
        }

        /// <summary>
        /// Method to implement retry policy controlled by configuration
        /// </summary>
        /// <returns>Returns true if it's ok to retry</returns>
        public bool canRetry()
        {
            // Keep track of current retries
            currentTries++;
            Console.WriteLine($"Retrying: {currentTries}");

            // Use a delay if this isn't the first try
            if (currentTries != 1)
            {
                Thread.Sleep(int.Parse(configuration["delay"]));
            }

            if (currentTries < int.Parse(configuration["number-of-retries"])) {
                return true;
            } else {
                return false;
            }
        }

        public void reset()
        {
            currentTries = 0;
        }
    }
}